kong/runloop/handler.lua

```diff
@@ -687,12 +687,26 @@ return {
   },
   header_filter = {
     before = function(ctx)
+      local var = ngx.var
+      local header = ngx.header
+
       if ctx.KONG_PROXIED then
         local now = get_now()
         -- time spent waiting for a response from upstream
         ctx.KONG_WAITING_TIME             = now - ctx.KONG_ACCESS_ENDED_AT
         ctx.KONG_HEADER_FILTER_STARTED_AT = now
       end
+
+      local upstream_header = constants.HEADERS.UPSTREAM_STATUS
+      if singletons.configuration.headers[upstream_header] then
+        local matches, err = ngx.re.match(var.upstream_status,
+                                          "[0-9]+$", "oj")
+        if err then
+          log(ERR, "failed to set upstream status header", err)
```

Comment:
This message is wrongly formatted. It is also somewhat obscure. It should be better formatted and cold be clarified, like so:
```lua
log(ERR, "failed to set ", upstream_status_header, " header: ", err)
```

(Notice the more appropriate name for the `upstream_header` variable as well...)